framework/src/modules/chain/logic/block/block_bft.js

```diff
@@ -0,0 +1,573 @@
+	 * @param {block} block
+	 * @throws {Error}
+	 * @returns {boolean} Verified hash, signature and publicKey
+	 * @todo Add description for the params
+	 */
+	verifySignature(block) {
+		const signatureLength = 64;
+		let res;
+
+		try {
+			const data = this.getBytes(block);
+			const dataWithoutSignature = Buffer.alloc(data.length - signatureLength);
+
+			for (let i = 0; i < dataWithoutSignature.length; i++) {
+				dataWithoutSignature[i] = data[i];
+			}
+			const hash = crypto
+				.createHash('sha256')
+				.update(dataWithoutSignature)
+				.digest();
+			const blockSignatureBuffer = this.scope.ed.hexToBuffer(
+				block.blockSignature
+			);
+			const generatorPublicKeyBuffer = this.scope.ed.hexToBuffer(
+				block.generatorPublicKey
+			);
+			res = this.scope.ed.verify(
+				hash,
+				blockSignatureBuffer || ' ',
+				generatorPublicKeyBuffer || ' '
+			);
+		} catch (e) {
+			throw e;
+		}
+
+		return res;
+	}
+
+	/**
+	 * Description of the function.
+	 *
+	 * @param {block} block
+	 * @throws {string|Error}
+	 * @returns {Object} Normalized block
+	 * @todo Add description for the function and the params
+	 */
+	objectNormalize(block) {
+		Object.keys(block).forEach(key => {
+			if (block[key] === null || typeof block[key] === 'undefined') {
+				delete block[key];
+			}
+		});
+		const report = this.scope.schema.validate(block, this.schema);
+		if (!report) {
+			throw new Error(
+				`Failed to validate block schema: ${this.scope.schema
+					.getLastErrors()
+					.map(err => err.message)
+					.join(', ')}`
+			);
+		}
+		const {
+			transactionsResponses,
+		} = modules.processTransactions.validateTransactions(block.transactions);
+		const invalidTransactionResponse = transactionsResponses.find(
+			transactionResponse => transactionResponse.status !== TransactionStatus.OK
+		);
+		if (invalidTransactionResponse) {
+			throw invalidTransactionResponse.errors;
+		}
+
+		return block;
+	}
+
+	/**
+	 * Binds scope.modules to private variable modules.
+	 */
+	// eslint-disable-next-line class-methods-use-this
+	bindModules(__modules) {
+		modules = {
+			processTransactions: __modules.processTransactions,
+		};
+	}
+
+	/**
+	 * Description of the function.
+	 *
+	 * @param {block} block
+	 * @throws {Error}
+	 * @returns {!Array} Contents as an ArrayBuffer
+	 * @todo Add description for the function and the params
+	 */
+	// eslint-disable-next-line class-methods-use-this
+	getBytes(block) {
+		const capacity =
+			4 + // version (int)
+			4 + // timestamp (int)
+			4 + // height (int)
+			4 + // maxHeightPreviouslyForged (int)
+			4 + // prevotedConfirmedUptoHeight (int)
+			8 + // previousBlock
+			4 + // numberOfTransactions (int)
+			8 + // totalAmount (long)
+			8 + // totalFee (long)
+			8 + // reward (long)
+			4 + // payloadLength (int)
+			32 + // payloadHash
+			32 + // generatorPublicKey
+			64 + // blockSignature or unused
+			4; // unused
+		let bytes;
+		try {
+			const byteBuffer = new ByteBuffer(capacity, true);
```

Comment:
ByteBuffer library is actually not needed.
Please use the native buffer.
Original code will be changed in https://github.com/LiskHQ/lisk-sdk/issues/3426

Reply:
#3807 issue responsible for this change